src/from/dynamodb.js

```diff
@@ -22,8 +27,8 @@ export const fromDynamodb = (event) => // eslint-disable-line import/prefer-defa
         record,
         event: {
           id: record.eventID,
-          type: `${calculateEventTypePrefix(record)}-${calculateEventTypeSuffix(record)}`,
-          partitionKey: record.dynamodb.Keys.pk.S,
+          type: `${calculateEventTypePrefix(record, opt)}-${calculateEventTypeSuffix(record)}`,
```

Comment:
If you decide the change above is sensible this will also be required

```suggestion
          type: `${calculateEventTypePrefix(record, { skFn, discriminatorFn, eventTypePrefix })}-${calculateEventTypeSuffix(record)}`,
```